sw/host/opentitanlib/src/test_utils/rpc.rs

```diff
@@ -58,19 +59,34 @@ impl<T: DeserializeOwned> UartRecv for T {
                 let cap = console
                     .captures(ExitStatus::ExitSuccess)
                     .expect("RESP_OK capture");
-                let s = cap.get(1).expect("RESP_OK group").as_str();
-                Ok(serde_json::from_str::<Self>(s)?)
+                let Some((json_str, crc_str)) = cap.get(1).map(|m| m.as_str()).zip(cap.get(2).map(|m| m.as_str())) else {
+                    return Err(anyhow!("Malformed regex. Expected capture groups not present."));
```

Comment:
Please use a structured error here (and below) rather than an `anyhow!(...)`.

If there isn't an existing error appropriate for this condition, you can create one easily with the `thiserror` crate:
```
use thiserror::Error;

#[derive(Error, Debug)]
pub enum MySpecialError {
    #[error("Malformed message: expected capture groups not present")]
    MalformedMessage,
}
```

Reply:
It's not the message that is malformed but the expression. If the expression matches it should have two capture groups, so this error condition shouldn't be reached unless the expression is modified to have fewer or optional capture groups.

Reply:
This being the case, does it still warrant it's own error message or is it more similar to https://github.com/lowRISC/opentitan/blob/8ff4837913494f2aa310fc999d0f82cef0d8aff0/sw/host/opentitanlib/src/test_utils/rpc.rs#L73

Reply:
:+1: This sounds sensible. Done